releases/tex/master/extra/crowbar-upgrade

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+FILE="/root/crowbar-upgrade-`date +"%Y%m%d-%k%M%S"`.tar.gz"
+
+tmpdir=$(mktemp -d)
+pushd $tmpdir >/dev/null
+
+echo "Start knife backup..."
+mkdir "knife"
+knife backup export data_bags clients nodes roles -D knife 1>/dev/null
+echo "Knife backup done..."
+
+echo "Prepair crowbar backup..."
+cp -a /etc/resolv.conf{,.forwarders}
+sed -i -e "/^\s*nameserver/d" /etc/resolv.conf.forwarders
+for ns in `perl -ne 'BEGIN{$/=undef}; m/forwarders\s*{([0-9:;\[\]\.\s]+)}/; $f=$1||""; $f=~s/\s*//g; foreach $fw (split(";",$f)){print "$fw\n"}' /etc/bind/named.conf`; do
+    echo "nameserver $ns" >> /etc/resolv.conf.forwarders
+done
+
+pushd /opt/dell/crowbar_framework > /dev/null
+RAILS_ENV=production bin/rake db:data:dump
+mv db/data.yml db/production.yml
+popd > /dev/null
+
+mkdir -p "crowbar"
+source /opt/dell/crowbar_framework/config/crowbar.env
+echo $CROWBAR_VERSION > crowbar/version
+
+echo "Start crowbar backup..."
+ETCFILES="/etc/crowbar/crowbar.json /etc/crowbar/network.json /etc/crowbar.install.key /etc/resolv.conf.forwarders /etc/ntp.conf /etc/HOSTNAME /etc/hosts /etc/chef/*.pem"
+mkdir -p "crowbar/etc"
+for file in $ETCFILES; do
+  cp -r $file crowbar/etc/
```

Comment:
we might want to make sure that we use -p or -a in copy to not accidentally loosen the file permissions